### ISSUE TITLE
梯度下降

### ISSUE THREAD

**Issue by lfeng14** [lfeng14/leetccode#21]
#### 题目
<img width="574" alt="image" src="https://github.com/lfeng14/leetccode/assets/144297355/43f626af-c98d-4dfc-abd6-d408ed2b19c1">

#### 思路
- 往梯度相反方向前进可以达到局部极小值或者局部极大值；y = f(x) = x^2；
- 梯度下降法（英语：Gradient descent）是一个一阶[最优化](https://zh.wikipedia.org/wiki/%E6%9C%80%E4%BC%98%E5%8C%96)[算法](https://zh.wikipedia.org/wiki/%E7%AE%97%E6%B3%95)，通常也称为最陡下降法，但是不该与近似积分的最陡下降法（英语：Method of steepest descent）混淆。 要使用梯度下降法找到一个函数的[局部极小值](https://zh.wikipedia.org/wiki/%E6%9C%80%E5%80%BC)，必须向函数上当前点对应[梯度](https://zh.wikipedia.org/wiki/%E6%A2%AF%E5%BA%A6)（或者是近似梯度）的反方向的规定步长距离点进行[迭代](https://zh.wikipedia.org/wiki/%E8%BF%AD%E4%BB%A3)搜索。如果相反地向梯度正方向迭代进行搜索，则会接近函数的[局部极大值](https://zh.wikipedia.org/wiki/%E6%9C%80%E5%80%BC)点；这个过程则被称为梯度上升法。
- 源码
```
class Solution:
    def get_minimizer(self, iterations: int, learning_rate: float, init: int) -> float:
        minimizer = init

        for _ in range(iterations):
            derivative = 2 * minimizer
            minimizer = minimizer - learning_rate * derivative

        return round(minimizer, 5)
```

#### 参考
- [neetcode题目](https://neetcode.io/problems/gradient-descent)
